modules/persistent-storage-csi-vsphere-top-aware.adoc

```diff
@@ -0,0 +1,171 @@
+
+:FeatureName: vSphere topology Awareness
+include::snippets/technology-preview.adoc[leveloffset=+1]
+
+.Procedure
+To deploy {product-title} for vSphere storage on different zones and regions:
+
+. In the VMware vCenter vSphere client GUI:
+.. Create and assign zone and region type tags as needed to the hosts on the desired clusters.
+.. Create and apply "cat-zone-local-datastore" tags to the desired datastores.
+
++
+For more information about vSphere tags, see the WMware vSphere documentation.
+
+. In the `clustercsidriver` object YAML file, edit the "driverConfig" section:
+* Specify the zone and region tags that you created earlier.
+* Set "driverType" to "vSphere".
++
+[source, terminal]
+----
+~ $ oc get clustercsidriver csi.vsphere.vmware.com -o yaml
+----
++
+Example output:
++
+[source, terminal]
+----
+apiVersion: operator.openshift.io/v1
+kind: ClusterCSIDriver
+metadata:
+ name: csi.vsphere.vmware.com
+spec:
+ logLevel: Normal
+ managementState: Managed
+ observedConfig: null
+ operatorLogLevel: Normal
+ unsupportedConfigOverrides: null
+  driverConfig: 
+    driverType: vSphere <1>
+    vSphere:
+      topologyCategories: <2>
+        - k8s-zone
+        - k8s-region
+----
+<1> Ensure that "driverType" is set to "vSphere".
+<2> Zone and region tags created earlier in vCenter.
+
+. Verify that `CSINode` object has topology keys:
++
+[source, terminal]
+----
+~ $ oc get csinode
+----
++
+Example output:
++
+[source, terminal]
+----
+NAME                        DRIVERS     AGE
+co8-4s88d-infra-2m5vd       1           27m
+co8-4s88d-master-0          1           70m
+co8-4s88d-master-1          1           70m
+co8-4s88d-master-2          1           70m
+co8-4s88d-worker-j2hmg      1           47m
+co8-4s88d-worker-mbb46      1           47m
+co8-4s88d-worker-zlk7d      1           47m
+----
++
+[source, terminal]
+----
+~ $ oc get csinode co8-4s88d-worker-j2hmg -o yaml
+----
++
+Exmaple output:
++
+[source, terminal]
+----
+...
+spec:
+    drivers:
+    - allocatable:
+        count: 59
+    name: csi-vsphere.vmware.com
+    nodeID: co8-4s88d-worker-j2hmg
+    topologyKeys: <1>
+    - topology.csi-vmware.com/k8s-zone
+    - topology.csi-vmware.com/k8s-region
+----
+<1> Topology keys from vSphere zone and region catagory tags.
++
+[NOTE]
+=====
+`CSINode` objects may take some time to receive updated topology information. After the driver is updated, `CSINode` objects should have topology keys in them.
+=====
+
+. Create a storage policy that uses the category tags:
+.. In vCenter, from the main menu, click *Policies and Profiles*.
+.. On the *Policies and Profiles* page, in the left pane, click *VM Storage Policies*.
+.. Click *CREATE*.
+.. Type a name for the storage policy.
+.. For the rules, select tags that target the desired datastore.
+
+. Create a new storage class that uses the new zoned storage policy:
+.. Click *Storage* > *StorageClasses*.
```

Comment:
Here, we go back to letters (above we did bullets and letters). Maybe it was intentional?

Reply:
letters for sublevel ordered steps and bullets for unordered steps.